File: frontend/src/types/timer.ts
export type TimerStatus = 'active' | 'stopped'

export type TimerViewStatus =
  | 'active'
  | 'warning'
  | 'stopped'
  | 'signal'
  | 'completed'

export interface Timer {
  id: string
  title: string
  description: string
  imageUrl?: string
  durationSeconds: number
  minDurationSeconds: number | null
  timeShiftSeconds: number
  startedAt: string
  lastRunBy: string
  lastRunByDisplayName?: string | null
  status: TimerStatus
  soundEnabled: boolean
  createdAt: string
  updatedAt: string
}

export interface TimerFormPayload {
  title: string
  description: string
  durationMinutes: number
  minDurationMinutes: number | null
  imageFile: File | null
  removeImage?: boolean
}

File: backend/app/schemas/timer.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class TimerResponse(BaseModel):
    id: str

    title: str
    description: str
    imageUrl: str | None

    durationSeconds: int
    minDurationSeconds: int | None
    timeShiftSeconds: int

    startedAt: datetime
    lastRunBy: str | None

    status: Literal["active", "stopped"]

    createdAt: datetime
    updatedAt: datetime

    soundEnabled: bool = True


class TimerRestartRequest(BaseModel):
    timeShiftSeconds: int = 0